pkg/cloud/gcp/actuators/machine/machine_scope_test.go

```diff
@@ -0,0 +1,445 @@
+					t.Errorf("expected to provider id to be %s, got: %s", expectedProviderID, scope.providerID)
+				}
+			}
+		})
+	}
+}
+
+func TestPatchMachine(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "..", "..", "..", "config", "crds")},
+	}
+
+	cfg, err := testEnv.Start()
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cfg).ToNot(BeNil())
+	defer func() {
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}()
+
+	k8sClient, err := client.New(cfg, client.Options{})
+	g.Expect(err).ToNot(HaveOccurred())
+
+	testNamespaceName := "test"
+
+	testNamespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testNamespaceName,
+		},
+	}
+	g.Expect(k8sClient.Create(ctx, testNamespace)).To(Succeed())
+	defer func() {
+		g.Expect(k8sClient.Delete(ctx, testNamespace)).To(Succeed())
+	}()
+
+	userDataSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      userDataSecretName,
+			Namespace: defaultNamespaceName,
+		},
+		Data: map[string][]byte{
+			userDataSecretKey: []byte("userDataBlob"),
+		},
+	}
+
+	g.Expect(k8sClient.Create(ctx, userDataSecret)).To(Succeed())
+	defer func() {
+		g.Expect(k8sClient.Delete(ctx, userDataSecret)).To(Succeed())
+	}()
+
+	credentialsSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      credentialsSecretName,
+			Namespace: defaultNamespaceName,
+		},
+		Data: map[string][]byte{
+			credentialsSecretKey: []byte("{\"project_id\": \"test\"}"),
+		},
+	}
+
+	g.Expect(k8sClient.Create(ctx, credentialsSecret)).To(Succeed())
+	defer func() {
+		g.Expect(k8sClient.Delete(ctx, credentialsSecret)).To(Succeed())
+	}()
+
+	failedPhase := "Failed"
+
+	providerStatus := &v1beta1.GCPMachineProviderStatus{}
+
+	machineName := "test"
+	machineKey := types.NamespacedName{Namespace: testNamespaceName, Name: machineName}
+
+	testCases := []struct {
+		name   string
+		mutate func(*machinev1.Machine)
+		expect func(*machinev1.Machine) error
+	}{
+		{
+			name: "Test changing labels",
+			mutate: func(m *machinev1.Machine) {
+				m.Labels["testlabel"] = "test"
+			},
+			expect: func(m *machinev1.Machine) error {
+				if m.Labels["testlabel"] != "test" {
+					return fmt.Errorf("label \"testlabel\" %q not equal expected \"test\"", m.ObjectMeta.Labels["test"])
+				}
+				return nil
+			},
+		},
+		{
+			name: "Test setting phase",
+			mutate: func(m *machinev1.Machine) {
+				m.Status.Phase = &failedPhase
+			},
+			expect: func(m *machinev1.Machine) error {
+				if m.Status.Phase != nil && *m.Status.Phase == failedPhase {
+					return nil
+				}
+				return fmt.Errorf("phase is nil or not equal expected \"Failed\"")
+			},
+		},
+		{
+			name: "Test setting provider status",
+			mutate: func(m *machinev1.Machine) {
+				instanceID := "123"
+				instanceState := "running"
+				providerStatus.InstanceID = &instanceID
+				providerStatus.InstanceState = &instanceState
```

Comment:
This is using a "global", shouldn't this be setting the providerStatus on the particular machine itself? Otherwise it could interfere with other tests?